### DISCUSSION TITLE
How to override authorization_endpoint in .well-known/openid-configuration ?

### DISCUSSION THREAD

**Discussion by psychowood** [authelia/authelia#7717] [Q&A] [answered]
Hi,
In my configuration I have authelia in two different networks:

- one public, behind a reverse proxy, used for browser access and **not accessible to docker containers** - e.g. https://authelia.public
- one private, to allow communication between containers - e.g. http://authelia:9091

When configuring an application manually, I can specify the correct private/public urls:

<img width="623" alt="image" src="https://github.com/user-attachments/assets/6e2583dc-b359-4e53-84d5-020b44652d48">

but I'm having issues with [Gitea](https://www.authelia.com/integration/openid-connect/gitea/), that uses the `/.well-known/openid-configuration` alone:

- If I use the public url `https://authelia.public/.well-known/openid-configuration` it is not working because (as expected) is not reachable
- If I use the private url `http://authelia.lan.used.to.it/.well-known/openid-configuration` it seems to work, but then the browser is redirected to the internal url for authorization `http://authelia:9091/api/oidc/authorization?client_id=gitea&.....` that's not public.

Is there a way to force a specific authorization_endpoint in `/.well-known/openid-configuration` ?

Thanks in advance

**Comment by james-d-elliott** [comment, marked as answer]
To be absolutely clear we do not offer any formal support for any of this. The endpoints the users access are the same endpoints a backend must access, any hack to get around this architecture issue is almost guaranteed to break in the future. It should also be absolutely clear that several of the available endpoints **_MUST_** use `https://`  per the spec (including the token endpoint) and this is a minimum supported configuration for authelia, you are likely to run into this issue as well.

The objectively correct solution is to have Gitea be able to communicate over the public URL to authelia. This is supported by numerous sections of the relevant specifications. We made all decisions around this intentionally with the spec in mind. 

However, we determine the values of the well-known endpoint using the required headers. If these were changed it theoretically could hack around this situation in a very fragile way.

**Reply by psychowood** [reply]
Thanks for the explanation. I wasn't aware of the https requirements, I'll work around enabking the missing routes. Right know I disabled OpenID and I'm using the remote-user http header.

**Reply by james-d-elliott** [reply]
If you're using docker there are simple methods to get this to work here https://www.authelia.com/integration/openid-connect/frequently-asked-questions/#configure-dns-appropriately
